src/tests/system/tests/test_ad.py

```diff
@@ -111,3 +111,78 @@ def test_authentication__using_the_users_email_address(client: Client, ad: AD, m
     assert client.auth.parametrize(method).password(
         "uSEr_3@alias-dOMain.com", "Secret123"
     ), "User uSEr_3@alias-dOMain.com failed login!"
+
+
+@pytest.mark.topology(KnownTopology.AD)
+@pytest.mark.ticket(jira="RHEL-75484")
+@pytest.mark.importance("high")
+def test_range_retrieval__group_membership(client: Client, ad: AD):
+    """
+    :title: Users with with more groups than MaxValRange can be retrieved
```

Comment:
@jakub-vavra-cz, could you please also add a test for a case "a group with more than MaxValRange users"?

Reply:
Ok, then for a start, I need to understand something.

Does AD really store group membership in user objects directly?

IIUC, that's not the case with OpenLDAP/DS (at least by default).